elfi/methods/utils.py

```diff
@@ -184,29 +184,72 @@ def logpdf(cls, x, means, cov=1, weights=None):
         return np.log(cls.pdf(x, means=means, cov=cov, weights=weights))
 
     @classmethod
-    def rvs(cls, means, cov=1, weights=None, size=1, random_state=None):
+    def rvs(cls, means, cov=1, weights=None, size=1, prior_logpdf=None,
+            max_trials=100, random_state=None):
         """Draw random variates from the distribution.
 
         Parameters
         ----------
         means : array_like
             means of the Gaussian mixture components
-        weights : array_like
-            1d array of weights of the gaussian mixture components
-        cov : array_like
+        cov : array_like, optional
             a shared covariance matrix for the mixture components
-        size : int or tuple
+        weights : array_like, optional
+            1d array of weights of the gaussian mixture components
+        size : int or tuple, optional
+        prior_logpdf : callable, optional
+            Can be used to check validity of random variable.
+        max_trials : int, optional
+            Maximum number of trials to find allowed random variables.
         random_state : np.random.RandomState or None
 
         """
-        means, weights = cls._normalize_params(means, weights)
         random_state = random_state or np.random
+        means, weights = cls._normalize_params(means, weights)
 
-        inds = random_state.choice(len(means), size=size, p=weights)
-        rvs = means[inds]
-        perturb = ss.multivariate_normal.rvs(
-            mean=means[0] * 0, cov=cov, random_state=random_state, size=size)
-        return rvs + perturb
+        # additional complexity due to supporting scipy-like ambiguity of arguments
```

Comment:
Could you elaborate on this? I'd be interested to know what did not work in the implementation above.

Reply:
Oh, so the comment didn't refer to the previous implementation, but to the needed `if` construct.

Reply:
Yes, it's a bit unclear to me what is  achieved with the added if-else construct. The comment in the code doesn't really help :)